src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+        that.addEditToUnsavedList("Enabled Product for Context: " + product);
+    }
+    else {
+        that.applier.change(path, false, "DELETE");
+        that.addEditToUnsavedList("Un-enabled Product for Context: " + product);
+    }
+    that.applier.change("npsetApplications", gpii.devpmt.npsetApplications(that.model.flatPrefs));
+    that.applier.change("unsavedChangesExist", true);
+};
+
+/**
+ * Event Listener for when a button is clicked to delete
+ * a context. Fetches the context to delete from the elements
+ * data-contextid attribute.
+ *
+ * This listener doesn't actually delete the context, but
+ * opens the confirm delete context dialog.
+ *
+ * @param event (Event) Browser event object
+ */
+gpii.devpmt.onDeleteContext = function (that, event) {
+    var contextId = event.currentTarget.dataset.contextid;
+    that.applier.change("activeModalDialog.contextId", contextId);
+    that.events.openConfirmDeleteContextDialog.fire();
+};
+
+gpii.devpmt.editValueEvent = function (that, event) {
+    var newCurrent = {
+        context: event.currentTarget.dataset.context,
+        term: event.currentTarget.dataset.term,
+        //value: event.currentTarget.dataset.value,
+        product: event.currentTarget.dataset.product
+    };
+
+    var newMetadata = {};
+    if (!newCurrent.product) {
+        // Common Term
+        newMetadata = that.options.commonTerms[newCurrent.term];
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupGenericPrefValue(newCurrent.context, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    else {
+        // Application Specific
+        console.log(newCurrent);
+        // TODO ontology
+        newMetadata = gpii.devpmt.findProductSettingMetadata(that.model.allSolutions,
+            newCurrent.product.slice(38), newCurrent.term);
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupProductPrefValue(newCurrent.context, newCurrent.product, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    that.applier.change("currentlyEditing.current", newCurrent);
+    that.applier.change("currentlyEditing.metadata", newMetadata);
+    that.applier.change("currentlyEditing.active", true);
+    that.applier.change("unsavedChangesExist", true);
+    that.prefsAdjuster.renderInitialMarkup();
+    // $(document).foundation();
+    // https://github.com/zurb/foundation-sites/issues/7899
+    $(".sticky").foundation("_calc", true);
+};
+
+
+/**
+ * Temporary workaround until we set up an ajax feed for these. The server
+ * side template is rendering the transformed preferences in another script
+ * element at the top of the page.
+ */
+gpii.devpmt.npsetInit = function (that) {
+    // Should use change applier
```

Comment:
Must use ChangeApplier. But many of these are fished statically out of options and so can just be referenced in the initial model from there.

Reply:
@amb26 I understand that the changeApplier should be used, and that I'm fishing these out of the options, but I'm not sure what `and so can just be referenced in the initial model from there.` means.  Does this mean something different that just using the change applier here and passing in the value that is taken from the options?  I'm not sure where `there` is.

Reply:
everything is using the change applier